adb/sign_m2crypto.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+
+from M2Crypto import RSA
+
+import adb_protocol
+
+
+class M2CryptoSigner(adb_protocol.AuthSigner):
+  """AuthSigner using M2Crypto."""
+
+  def __init__(self, rsa_key_path):
```

Comment:
Having a different signature is fine since the customer instantiates this or the other class and passes that in.

If you want to support passing in the keys in separately, you can create a subclass that doesn't call super's \__init__ and just sets self.public_key and self.rsa_key from arguments itself.